### ISSUE TITLE
borg mount -o [options] does not take some FUSE options (gid, umask)

### ISSUE THREAD

**Issue by lauhayden** [borgbackup/borg#3507]
I would like to mount a Borg repo with some given permissions - particularly, for it to be readable by a group. I can successfully mount with `allow-other`, which mounts with 744 permissions, but I don't want the files to be readable by other users. Both the options that fail are documented in [fuse(8)](http://man7.org/linux/man-pages/man8/mount.fuse.8.html). 
```
hayden@----:~$ sudo borg mount -o gid=[unix group] [repo location] [mount point]
Enter passphrase for key [repo location]:
fuse: unknown option `gid=[unix group]'

hayden@----:~$ sudo borg mount -o umask=027 [repo location] [mount point]
Enter passphrase for key [repo location]:
fuse: unknown option `umask=027'```
```

**Comment by ThomasWaldmann** [comment]
Please check your console example, it looks like you edited it a bit too much:
- 1st: group_id= or gid= ?
- 2nd: there is no umask=

I looked at the man page, but the description there is rather short, so I'm unsure what it should do precisely. Also, in case they override uid and mode of the files in the borg archive, that can be a information loss / problematic in other ways.

**Comment by lauhayden** [comment]
Derp, whoops. I've tried both `group_id` and `gid`, and neither group names nor numbers work. `umask` does not work either. 

I was assuming that the options were passed through by Borg, and interpreted by FUSE?  In any case, I expected behaviour similar to [`ntfs-3g`](https://www.tuxera.com/community/ntfs-3g-manual/#6) with the `gid` and `umask` options (whether that is a reasonable assumption or not, I'm not sure). `gid` does not change the owner of the actual NTFS files, but changes the group owner of the mounted view. Same thing with `umask`.

**Comment by Mathiasdm** [comment]
I'm seeing the same problem: 'uid=...', 'gid=...' and 'umask=...' result in 'fuse: unknown option', even though 'man fuse' shows these options.

I'm using CentOS 7, if that makes any difference.

**Comment by ThomasWaldmann** [comment]
Borg uses the original uid/gid as stored in the archive for the filesystem objects shown in the fuse mount.

So, do I understand it correctly, that supporting uid/gid/umask for fuse mounts means to change uid/gid/mode of the filesystem objects shown in the fuse mount?

If the backup archive is from the current host (or a host with the same user/group-to-uid/gid mapping), that basically means giving access to items where the user/group did not have access normally. Similar for other mapping, but there the permissions are unclear anyway (besides normal-user != root).

From a security perspective, this is maybe not a problem: if a user is able to read some borg archive, he can fully extract it anyway (as user [then all files will be owned by that user] or as root [keeping original user/uid/group/gid]).

So this boils down to "don't give not fully trusted users access to your backup archives".
We would just extend that to "don't give not fully trusted users access to your borg fuse mount".

**Comment by Mathiasdm** [comment]
Well, to me, the exact requirement would be: the result of running 'borg mount' should look the same as running 'borg extract' (as user).

**Comment by Mathiasdm** [comment]
Ah, interesting. In that case, I might be able to solve it myself (for my use case) by putting an overlayfs on top of borg mount and doing the chown/chmod there.

**Comment by ThomasWaldmann** [comment]
See #3769 - would be good if this would get some practical testing.

**Comment by ThomasWaldmann** [comment]
fixed by #3843 (forward port of #3769).
